xl2times/dd_to_csv.py

```diff
@@ -55,20 +55,19 @@ def parse_parameter_values_from_file(
             param_data = []
             # Parse values until a line with / is encountered.
             while not data[index].startswith("/") and data[index] != "":
-                words = data[index].split(" ")
+                line = data[index]
 
                 # Either "value" for a scalar, or "key value" for an array.
-                if len(words) == 1:
-                    attributes = []
-                elif len(words) == 2:
-                    attributes = words[0].split(".")
-                    attributes = [a if " " in a else a.strip("'") for a in attributes]
+                # So value is always the last word, or only token
+                split_point = line.rfind(" ")
+                if split_point == -1:
+                    # if only one word
+                    attributes, value = [], line
                 else:
-                    raise ValueError(
-                        f"Unexpected number of spaces in parameter value setting: {data[index]}"
-                    )
+                    attributes, value = line[:split_point], line[split_point + 1 :]
+                    attributes = attributes.split(".")
+                    attributes = [a if " " in a else a.strip("'") for a in attributes]
```

Comment:
The `rfind()` works more reliably, on the assumption that there's always a space before the value, so it'll work with `'key with spaces' value` style strings

Reply:
For example, here's a line from one of our DD files that (i think) would have triggered the ValueError, because of the spaces in `UC_non-decreasing EE penetration`:

`'UC_non-decreasing EE penetration'.RHS.'QLD'.2015.'EE2_Pri_Pub-b'.ANNUAL 1`